src/pybind/mgr/cephadm/module.py

```diff
@@ -844,6 +845,33 @@ def _prepare_host(self, host, addr=None):
                     self.event.set()
         return 0, '%s (%s) ok' % (host, addr), err
 
+    @orchestrator._cli_read_command(
+        'cephadm generate-service-specs',
+        desc='Generates a YAML description of existing daemons without services')
+    def _generate_service_specs(self):
+        """
+        When doing an adoption from an existing cluster, the daemons get adopted, but
+        no sevice specs are generated. Leaving cephadm in an odd state.
+
+        This command cannot be properly implemented, cuase of:
+
+        * we don't know the exact placement specifier for MONs
+        * specs can't be deduced. Like port numbers, passwords etc.
```

Comment:
I think that maybe we should create a fixed "template" spec file for each daemon/service, including all the attributes we need to configure the daemon/service.
This will make easy to "adopt" things.. because we will know what attributes we are going to need, what are the defaults and if there is a customization in any attribute.

Reply:
you mean like e.g.

```yaml
service_type: rgw
...
spec:
#   rgw_port: <adapt this, if required>
```
?

Reply:
yes

Reply:
liek so? https://github.com/ceph/ceph/pull/35669/files#diff-bb6c1a9ce7cb5838b1e8cb287bf54412

Reply:
ok, turned out, having this this made this PR much more complicated and thus not sustainable.